### PR TITLE
ci: set up yarn berry and gh workflows

### PR DESCRIPTION
**Pull request by mcataford**
## Description

This sets up Yarn berry and a standard test/lint/build/release workflow in GH actions. The website steps are intentionally omitted since the website is currently blank.

__This changes the publish flow__: you now need to create a new tag, which will trigger the release process.

## QA:

- Ran the `pull_request` workflow with `on: pull_request` so that it would run on this branch, verified it passed.
- Set up a local registry via `verdaccio` [sic] and verified that `yarn lerna-publish` works with yarn berry.

`CODECOV_TOKEN` and `NPM_PUBLISH_TOKEN` were also set up as repo secrets.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,6 +27,7 @@
   "scripts": {
     "lint": "eslint --ignore-path .gitignore .",
     "lint-fix": "eslint --ignore-path .gitignore . --fix",
+    "test": "",
```
Comment (mcataford) [inline package.json:30]: Intentionally empty, will be filled in once @ju-nc is done with his testing PR.

File: .yarnrc.yml [hunk 2]
```diff
@@ -0,0 +1 @@
+yarnPath: ".yarn/releases/yarn-berry.cjs"
```
Comment (noahnu) [inline .yarnrc.yml:1]: This is fine, but as an alternative you can do: `yarn set version latest` to pin the very to a specific version number.

File: package.json [hunk 3]
```diff
@@ -1,6 +1,5 @@
 {
   "name": "root",
-  "private": false,
```
Comment (noahnu) [inline package.json:3]: ?

**Review by noahnu — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (mcataford) [reply package.json:3]: Remnant from our workflow experiments I think, let me take that out.

File: .yarnrc.yml (hunk 2 above)
Comment (mcataford) [reply .yarnrc.yml:1]: In this case would we just ignore the yarnrc file?

Comment (noahnu) [reply .yarnrc.yml:1]: No you have to include the yarnrc

**Comment by mcataford** [comment]
Removes the extraneous change to the package file and fixed a typo, your review got dismissed @noahnu 😓

**Review by noahnu — APPROVED** [review]

**Pull request merged**